frontend/packages/kubevirt-plugin/src/k8s/requests/cdi-upload/cdi-upload-requests.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable camelcase, @typescript-eslint/camelcase,no-await-in-loop */
+import { getName, getNamespace } from '@console/shared';
+import { EnhancedK8sMethods } from '../../enhancedK8sMethods/enhancedK8sMethods';
+import { DataVolumeModel, UploadTokenRequestModel } from '@console/kubevirt-plugin/src/models';
+import { V1alpha1DataVolume } from '@console/kubevirt-plugin/src/types/vm/disk/V1alpha1DataVolume';
+import { apiVersionForModel } from '@console/internal/module/k8s';
+import { delay } from '../../../utils/utils';
+import { K8sCreateError } from '../../enhancedK8sMethods/errors';
+
+const PVC_STATUS_DELAY = 2 * 1000;
+
+const UPLOAD_STATES = {
+  SCHEDULED: 'UploadScheduled',
+  READY: 'UploadReady',
+};
+
+const waitForUploadReady = async (dvName, namespace, k8sGet, counter = 20) => {
+  let dv;
+  for (let i = 0; i < counter; i++) {
+    if (dv?.status?.phase === UPLOAD_STATES.READY) {
+      return true;
+    }
+    await delay(PVC_STATUS_DELAY);
+    dv = await k8sGet(DataVolumeModel, dvName, namespace);
+  }
+
+  return false;
+};
+
+export const createUploadToken = async (pvcName: string, namespace: string) => {
+  const { k8sCreate } = new EnhancedK8sMethods();
+  const tokenRequest = {
+    apiVersion: apiVersionForModel(UploadTokenRequestModel),
+    kind: UploadTokenRequestModel.kind,
+    metadata: {
+      name: pvcName,
+      namespace,
+    },
+    spec: {
+      pvcName,
+    },
+  };
+
+  try {
+    const resource = await k8sCreate(UploadTokenRequestModel, tokenRequest);
+    return resource?.status?.token;
+  } catch (error) {
+    throw new K8sCreateError(error.message, tokenRequest);
+  }
+};
+
+export const createUploadPVC = async (dataVolume: V1alpha1DataVolume) => {
+  const { k8sCreate, k8sGet } = new EnhancedK8sMethods();
+  const dataVolumeName = getName(dataVolume);
+  const namespace = getNamespace(dataVolume);
+
+  try {
+    await k8sCreate(DataVolumeModel, dataVolume);
+    await waitForUploadReady(dataVolumeName, namespace, k8sGet);
+    const token = await createUploadToken(dataVolumeName, namespace);
+
+    return { token };
+  } catch (error) {
+    throw new K8sCreateError(error.message, dataVolume);
+  }
+};
+
+export const killUploadPVC = async (name: string, namespace: string) => {
+  const { k8sKill } = new EnhancedK8sMethods();
+  try {
+    await k8sKill(DataVolumeModel, { metadata: { name, namespace } });
+  } catch (error) {
+    throw new Error(error.message);
```

Comment:
no need to catch since you only wrap the error into another one. Or make the catch more meaningful :)